src/values/AbstractObjectValue.js

```diff
@@ -744,7 +746,22 @@ export default class AbstractObjectValue extends AbstractValue {
       return this.$SetPartial(P, V, Receiver);
     }
 
+    let realm = this.$Realm;
     let elements = this.values.getElements();
+
+    if (this.isSimpleObject()) {
+      // If this is a simple object, we know that we won't invoke any special
+      // logic so we can fast path it through the ordinary object model which
+      // will call $GetOwnProperty and $DefineOwnProperty on the Receiver.
+      // These will ensure that all possible values and descriptors are
+      // represented.
+      for (let cv of elements) {
+        invariant(cv instanceof ObjectValue);
+        return Properties.OrdinarySet(realm, cv, P, V, Receiver);
```

Comment:
Because this is a simple object this will always issue these two calls:

```js
let desc = OrdinaryGetOwnProperty(realm, Receiver, P);
// do some processing of the descriptor or create a new one
OrdinarySetOwnProperty(realm, Receiver, P, desc);
```

Note how it doesn't actually ever touch or need the `cv` object passed to it if it is simple. The OrdinaryGetOwnProperty gets a joined descriptor of all possible values on all the objects. It then updates all the possible objects with that joined value.

This is very obscure at best. Which is why I don't like this approach. I could just remove this whole thing and it'll just work. Another thing we could do is make a forked version of OrdinarySet's logic that by-passes checking the prototype chain for setters and such.

Reply:
I see. Just remove this.